### PR TITLE
Bonds view business logic #806

### LINKED ISSUE #806: Bonds view business logic

### PR DESCRIPTION
**Pull request by thatStrangeGuyThorchain**
Added:
- `UserNodes` context and `userNodes` service to store and manipulate user's saved nodes to watch
- business logic to load node's info for `BondsTable` component with caching previously loaded results

Refactor:
- moved `NodeInfo` to the `src/renderer/services/thorchain/types.ts`

Fix:
- fixed not reseting form values at the `BondsTable` component after submit

closes #806

### CODE REVIEW

File: src/renderer/services/storage/userNodes.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import { Address } from '@xchainjs/xchain-client'
+import * as A from 'fp-ts/Array'
+import { eqString } from 'fp-ts/Eq'
+import * as FP from 'fp-ts/function'
+import * as O from 'fp-ts/Option'
+import * as RxOp from 'rxjs/operators'
+
+import { getStorageState$, getStorageState, modifyStorage } from './storage'
+
+const userNodes$ = FP.pipe(
```
Comment (veado) [inline src/renderer/services/storage/userNodes.ts:10]: [nitpick] type ^ ?

File: src/renderer/services/thorchain/thorNode.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+              errorId: ErrorId.GET_NODE,
+              msg: `Failed to load info for ${node}`
+            })
+          )
+        )
+      )
+    ),
+    liveData.map((nodeData: NodeInfoResponse) => ({
+      bond: baseAmount(nodeData.bond),
+      award: baseAmount(nodeData.current_award),
```
Comment (veado) [inline src/renderer/services/thorchain/thorNode.ts:71]: [nitpick] ^ `baseAmount(..., THOR_DECIMAL)`

File: src/renderer/views/wallet/BondsView.tsx [hunk 3]
```diff
@@ -1,3 +1,92 @@
...
+      ),
+    [oClient]
+  )
+
+  const { network$ } = useAppContext()
+  const network = useObservableState<Network>(network$, DEFAULT_NETWORK)
+
+  const userNodes = useObservableState(userNodes$, [])
+
+  const nodeInfoCacheRef = useRef<Record<Address, ReturnType<typeof getNodeInfo$>>>({})
```
Comment (veado) [inline src/renderer/views/wallet/BondsView.tsx:51]: `ReturnType<typeof getNodeInfo$>>` -> `NodeInfoLD` ?

**Review by veado — APPROVED** [review]
Nice one! Just few nitpicks ...

File: src/renderer/services/thorchain/thorNode.ts (hunk 2 above)
Comment (thatStrangeGuyThorchain) [reply src/renderer/services/thorchain/thorNode.ts:71]: sure. will add. but 8 is a default value anyway :)

**Review by veado — APPROVED** [review]

**Pull request merged**